### DISCUSSION TITLE
Is it encouraged to install 2 instances on the same server?

### DISCUSSION THREAD

**Discussion by fedi-will** [mastodon/mastodon#23077] [Q&A] [answered]
I am currently trying to install 2 instances on the same machine (no Docker). In 2018 @Gargron said in [this issue](https://github.com/mastodon/mastodon/issues/7972):
> Yes, you can. Different DB_NAME, different REDIS_NAMESPACE, different PORT for web/streaming services, multiple server definitions in nginx.

I also followed instructions from [external web link](https://nfgworld.com/multiple-mastodon-instances-on-a-single-server/) mentioned in that issue. 

Turns out in 2022/23 it is a bit more complicated. 
I finally got stuck with [this](https://github.com/mastodon/mastodon/discussions/22989) and [this](https://github.com/mastodon/mastodon/discussions/22990) problem. 

I am not sure anymore if I should invest more time trying to find a solution.

Could someone from the team, like @nightpool?,  please give a simple Yes/No if it is still recommended to run 2 instances on same server?

**Comment by fedi-will** [comment]
Giving up...
Might not be possible or only by having having/solving some issues. 
See comments in [this stackoverflow question](https://stackoverflow.com/questions/75102278/rails-use-another-env-file)

**Reply by nightpool** [reply]
from the comments of that issue it sounds like you're trying to run two separate instances in the same *directory* simply by changing .env files. that seems slightly dangerous to me, since they could run into conflicts trying to upload files (if you're not using block storage) or log files, etc, besides the complications you're running into with .env files. why not just use two separate directories?

**Comment by nightpool** [comment, marked as answer]
Why wouldn't it be recommended? Servers are just machines, you can install as many mastodons on as many servers as you want. I run three servers on one machine without any issues.

**Comment by fedi-will** [comment]
I see. 
I followed instruction from [this post](https://nfgworld.com/multiple-mastodon-instances-on-a-single-server/) - and got into the described troubles. 
Thank you for your reply.
